File: frontend/types/blocked-company.ts
import type { Company } from "./job";

export interface BlockedCompany {
  id: number;
  user_id: number;
  company_id: number;
  created_at: Date;
  company: Company;
}

export interface BlockedCompanyListResponse {
  blocked_companies: BlockedCompany[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

File: backend/app/models/blocked_company.py
from __future__ import annotations

from sqlalchemy import Column, DateTime, ForeignKey, Integer, func
from sqlalchemy.orm import relationship

from app.db import Base


class BlockedCompany(Base):
    __tablename__ = "blocked_companies"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(
        Integer,
        ForeignKey("users.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    company_id = Column(
        Integer,
        ForeignKey("companies.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    created_at = Column(
        DateTime(timezone=True),
        nullable=False,
        server_default=func.now(),
    )

    user = relationship("User", back_populates="blocked_companies")
    company = relationship("Company", back_populates="blocked_companies")








